wordpress-mobile/WordPress-Android
# issue/62-simple post search

Fixes #62 

This PR adds the ability to search published and private sites on jetpack/wpcom sites.

**Implementation**
I used Pages Search functionality as a reference. Pages Search is using a dedicated search fragment, ViewModels, and adapters. I originally tried to go the same route but realized that this brings a lot of extra code that we will have to maintain alongside existing Post List, so I decided to try and use existing PostListFragment for search (huge mistake, but too late!)

The current implementation of Post List is pretty robust, and meant to work in a "fire and forget" kind of way, so I had to make a couple (maybe a lot) of minor changes to accommodate special behavior of Search list. I tried not to touch the core of PostListViewModel, so the changes are spread through a couple of helper classes and PostListFragment 

Search button logic in the toolbar is mostly copied from Pages List, with a couple of minor changes.

**UX**

- The search ignores filters in main post list and looks for Published and Private posts for all authors.
- Search list respects "view mode" selected at the main post list (condensed or full).
- Delay for search query input is 500 ms.
- One thing that was not covered in a mock was progress indication. Similar search screens we have in the app are doing local DB search, so there is almost no delay. Post search does a network search, and most of the time it's pretty fast, so really brief progress indication every time you stopped tapping was really annoying. To combat this I added a delay to PTR indicator, that appears if a result has not arrived within 1 second.


![search](https://user-images.githubusercontent.com/728822/59991705-0939c400-960e-11e9-9dca-edd88918cb6d.png)


To test:
Make sure the post Search button is not visible for self-hosted sites without a jetpack.
Make sure post search works as expected.

Update release notes:
